Lib/test/test_importlib/identity.py

```diff
@@ -0,0 +1,521 @@
+        while j >= 0 and not (stop := entries[i][j].category is quote):
+            token = entries[i][j]
+            if token.category not in type_candidates:
+                repls = ()
+            elif token.category is ident_name_other:
+                final = entries[i][j + 1] is ident_name_only
+                repls = get_name_qchar_idxs(token.value, not final)
+            else:
+                repls = range(len(token.value))
+            for k in repls:
+                index_candidates.append((i, j, k))
+            j -= 1
+        i -= 1
+    return index_candidates
+
+
+@st.composite
+def unbalance(draw, entries):
+    """
+    Unbalance quote characters in the list of entries without altering
+    the parsing of the list.
+    """
+    qchrs = draw(st.sets(st.sampled_from("\"'")))
+    index_candidates = unbalance_indexes(entries)
+    if not qchrs or not index_candidates:
+        return False
+    for qchr in qchrs:
+        idx_entry, idx_token, idx_char = draw(st.sampled_from(index_candidates))
+        token = entries[idx_entry][idx_token]
+        token.value = replace(token.value, idx_char, qchr)
+    return True
+
+
+# Non-leaf rules for "Author" and "Maintainer" fields:
+
+
+def name_entry(state):
+    return name_empty() | name_value(state)
+
+
+def name_empty():
+    return st.builds(Entry.from_empty, st.just(2), st.just(name_empty))
+
+
+@st.composite
+def name_value(draw, state):
+    choice = name_value_other(state) | quote(state)
+    sample = draw(st.lists(choice, min_size=1, max_size=10))
+    entry = Entry.from_empty(2, name_value)
+    entry[:1] = draw(merge_sub(state, sample, name_value_other, sub_entry))
+    return entry
+
+
+# Non-leaf rules for "Author-email" and "Maintainer-email" fields:
+
+
+def ident_entry(state):
+    return (
+        ident_empty()
+        | ident_name_only(state)
+        | ident_addr_only(state)
+        | ident_name_addr(state)
+    )
+
+
+def ident_empty():
+    return st.builds(Entry.from_empty, st.just(2), st.just(ident_empty))
+
+
+@st.composite
+def ident_name_only(draw, state):
+    entry = Entry.from_empty(2, ident_name_only)
+    entry[:1] = draw(ident_name(state, name_only=True))
+    return entry
+
+
+@st.composite
+def ident_addr_only(draw, state):
+    entry = Entry.from_empty(2, ident_addr_only)
+    addr = draw(ident_addr(state))
+    if draw(st.booleans()):
+        entry[1:] = [draw(ident_name_addr_delim("")), *addr]
+    else:
+        entry[2:] = addr
+    return entry
+
+
+@st.composite
+def ident_name_addr(draw, state):
+    name = draw(ident_name(state, name_only=False))
+    delim = draw(ident_name_addr_delim(name[-1].value))
+    addr = draw(ident_addr(state))
+    return Entry((*name, delim, *addr))
+
+
+@st.composite
+def ident_name_addr_delim(draw, name):
+    min_size = 0 if name.endswith("@") else 1
+    delim = draw(st.text(alphabet=" ", min_size=min_size, max_size=10))
+    return Token(ident_name_addr_delim, delim, True)
```

Comment:
In some cases like this one, you could avoid using `@st.composite` by returning a `st.builds()` strategy.  However others would be really annoying to express, and I think consistent style suggests that composite is fine here.